src/ooxml/java/org/apache/poi/xssf/streaming/SheetDataWriter.java

```diff
@@ -354,88 +355,54 @@ protected void outputQuotedString(String s) throws IOException {
             return;
         }
 
-        char[] chars = s.toCharArray();
-        int last = 0;
-        int length = s.length();
-        for (int counter = 0; counter < length; counter++) {
-            char c = chars[counter];
-            switch (c) {
-                case '<':
-                    writeLastChars(_out, chars, last, counter);
-                    last = counter + 1;
+        final String xa0 = Character.toString((char)0xa0);
+
+        for (Iterator<String> iter = new StringCodepointsIterable(s).iterator(); iter.hasNext();) {
```

Comment:
Can this be made into a for-each loop to hide the ugliness that is Java Iterators?